src/components/NurseryWithdrawals/PlantingProgressMap.tsx

```diff
@@ -92,29 +106,63 @@ export default function PlantingProgressMap({ plantingSiteId }: PlantingProgress
       );
       setRequestId(request.requestId);
       setFocusEntities([]);
+      setConfirmDeleteCb(() => () => null);
       setDispatching(true);
     },
     [dispatch]
   );
 
+  const updatePlantingComplete = useCallback(
+    (id: number, val: boolean) => {
+      if (!selectedZoneHasStats) {
+        completeUpdate(id, val)();
+      } else {
+        setConfirmDeleteCb(() => completeUpdate(id, val));
+        setStatsWarningDialogOpen(true);
```

Comment:
suggestion:
remove the callback approach, find the zone id here and set that as a property.
if the zone id is valid then show the modal (pass the zone id to the modal)
modal can send the zone id back as part of the onsubmit.
we can remove a bunch of additional state/message-passing